src/security_utilities_rust/src/microsoft_security_utilities_core/marvin.rs

```diff
@@ -2,22 +2,17 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 /// This is a Rust implementation of the Marvin32 checksum algorithm, the definitive native code for which is
-/// at https://github.com/microsoft/SymCrypt/blob/master/lib/marvin32.c. 
-use std::mem;
+/// at https://github.com/microsoft/SymCrypt/blob/master/lib/marvin32.c.
 
 /// Convenience method to compute a Marvin hash and collapse it into a 32-bit hash.
-pub fn compute_hash32(data: &[u8], seed: u64, offset: i32, length: i32) -> i32
-{
-    let hash64: i64 = compute_hash(data, seed, offset, length);
+pub fn compute_hash32(data: &[u8], seed: u64, offset: i32, length: i32) -> i32 {
```

Comment:
>compute_hash32 [](http://example.com/codeflow?start=7&length=14)

Johannes, thanks for all these contributions. Can we add some release notes for various performance improvements? How are we benchmarking these changes? For the C# code, we have a good repeatable mechanism for verifying perf gains, what's our Rust story.

Reply:
I will add some release notes later today :)

For the performance measurements, I'm using [`Criterion`][1]. I have written a few small benchmarks over on the [`users/jdraaijer/bench`][0] branch. It saves results and automagically compares the results from the previous run. Is this something that we may want to have a PR for so it can be merged and available to everyone?

[0]: https://github.com/microsoft/security-utilities/tree/users/jdraaijer/bench/src/security_utilities_rust
[1]: https://docs.rs/criterion/latest/criterion/

Reply:
@jdraaijer-microsoft Yes, it would be nice to have the benchmarks persisted, and documentation outlining how to repro.

Reply:
I've added a release note for the changes in this PR, and opened https://github.com/microsoft/security-utilities/pull/79 for the benchmarks.